api/model/src/main/java/org/projectnessie/model/MergeKeyBehavior.java

```diff
@@ -28,11 +33,61 @@ public interface MergeKeyBehavior {
 
   MergeBehavior getMergeBehavior();
 
+  /**
+   * If present, the current content on the target branch will be compared against this value.
+   *
+   * <p>This parameter is not supported when multiple commits will be generated, which means only
+   * "squashing" merge and transplant operations.
+   *
+   * <p>Supplying a {@link #getResolvedContent() resolved content} requires setting this attribute.
+   * The merge operation will result in a "conflict", if current value on the target branch is
+   * different from this value.
+   */
+  @JsonInclude(Include.NON_NULL)
+  @JsonView(Views.V2.class)
+  @Nullable
+  @jakarta.annotation.Nullable
+  Content getExpectedTargetContent();
+
+  /**
+   * Clients can provide a "resolved" content object, which will then automatically be persisted via
+   * the merge operation instead of detecting and potentially raising a merge-conflict, assuming the
+   * content-type is the same.
+   *
+   * <p>This functionality is not implemented for the "legacy" storage model, using this option with
+   * the "legacy" storage model will result in an error.
+   *
+   * <p>This parameter is not supported when multiple commits will be generated, which means only
+   * "squashing" merge and transplant operations.
+   *
+   * <p>It is mandatory to supply the {@link #getExpectedTargetContent() expected content value},
+   */
+  @JsonInclude(Include.NON_NULL)
+  @JsonView(Views.V2.class)
+  @Nullable
+  @jakarta.annotation.Nullable
+  Content getResolvedContent();
+
+  // TODO add metadata list from https://github.com/projectnessie/nessie/pull/6616
```

Comment:
Do we want to merge with this `TODO` or wait for #6616?

Reply:
It's a chicken-egg-problem neither can live without the other (wrt this TODO).